ophyd/device.py

```diff
@@ -968,6 +1006,29 @@ def describe_configuration(self) -> OrderedDictType[str, Dict[str, Any]]:
         """
         return self._describe_attr_list(self.configuration_attrs, config=True)
 
+    @property
+    def hints(self):
+        if self._hints is not None:
+            return self._hints
+        elif self._default_hints is None:
+            return {}
+        else:
+            return {'fields': [getattr(self, component_name).name
+                               for component_name in
+                                   self._default_hints['fields']]}
```

Comment:
This probably should be `self._default_hints.get('fields', [])` to be extra paranoid.

Reply:
do we want to do the read_attrs filtering / validation here as well?

Reply:
Done for the first comment. Which king of filtering would you suggest?